### PR TITLE
Improve logging in the Lambda Logs API handler

### PR DESCRIPTION
**Pull request by keisku**
### What does this PR do?

Don't ignore error details.

### Motivation

For better troubleshooting.

### Additional Notes

https://docs.aws.amazon.com/lambda/latest/dg/runtimes-logs-api.html#runtimes-logs-api-ref

### CODE REVIEW

File: pkg/serverless/logs/http.go [hunk 1]
```diff
@@ -32,10 +34,16 @@ func (l *LambdaLogsAPIServer) Close() {
...
+		w.WriteHeader(400)
+		return
+	}
 	defer r.Body.Close()
 	messages, err := parseLogsAPIPayload(data)
 	if err != nil {
+		log.Errorf("Error parsing Logs API payload: %s", err)
 		w.WriteHeader(400)
 	} else {
 		c.out <- messages
```
Comment (keisku) [inline pkg/serverless/logs/http.go:49]: If we can print the content of `messages` like [stream-logs](https://github.com/DataDog/datadog-agent/blob/33d43600c674a65c10336b0edadc2b677c6604a0/cmd/agent/subcommands/streamlogs/command.go#L56), it should give us a troubleshooting insight.

**Comment by keisku** [comment]
This error doesn't relate to this PR and should be handled in a different PR.
https://github.com/DataDog/datadog-agent/actions/runs/9219215650/job/25363974741?pr=25878
```
Run sudo yarn global add serverless@^3.36.0 --prefix /usr/local
yarn global v1.22.22
[1/[4](https://github.com/DataDog/datadog-agent/actions/runs/9219215650/job/25363974741?pr=25878#step:4:5)] Resolving packages...
warning serverless > @serverless/platform-client > querystring@0.2.1: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
warning serverless > aws-sdk > querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
warning serverless > aws-sdk > url > querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
warning serverless > json-refs > path-loader > superagent@7.1.[6](https://github.com/DataDog/datadog-agent/actions/runs/9219215650/job/25363974741?pr=25878#step:4:7): Please upgrade to v9.0.0+ as we have fixed a public vulnerability with formidable dependency. Note that v9.0.0+ requires Node.js v14.18.0+. See https://github.com/ladjs/superagent/pull/1800 for insight. This project is supported and maintained by the team at Forward Email @ https://forwardemail.net
warning serverless > @serverless/dashboard-plugin > yamljs > glob > inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
[2/4] Fetching packages...
error memoizee@0.4.16: The engine "node" is incompatible with this module. Expected version ">=.0.12". Got "1[8](https://github.com/DataDog/datadog-agent/actions/runs/9219215650/job/25363974741?pr=25878#step:4:9).20.2"
error Found incompatible module.
info Visit https://yarnpkg.com/en/docs/cli/global for documentation about this command.
Error: Process completed with exit code 1.
```

**Review by alexgallotta — APPROVED** [review]

**Review by git-thuerk-done — APPROVED** [review]
good for docs

**Review by purple4reina — APPROVED** [review]

**Pull request merged**